clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1219,6 +1220,9 @@ public void run() {
                 now = time.milliseconds();
                 handleResponses(now, responses);
             }
+
+            AppInfoParser.unregisterAppInfo(JMX_PREFIX, clientId, metrics);
```

Comment:
Perhaps some of this cleanup should go in a `finally`?

Reply:
Done. Created a new method for while loop and wrapped that up in try-finally.